### ISSUE TITLE
Should support nested objects in templates

### ISSUE THREAD

**Issue by WesTyler** [xogroup/toki-templater#1]
#### What are you trying to achieve or the steps to reproduce ?
```js
return Templater({ foo: '{{=it.foo}}' }, null, {
            context: {
                foo: {
                    bob: 'bar'
                }
            }
        })
            .then((result) => {

                expect(result).to.equal({ foo: { bob: 'bar' } });
            });
```

#### What result did you expect ?
Passing test.

#### What result did you observe ?
`Expected { foo: '[object Object]' } to equal specified value`

**Comment by lamchakchan** [comment]
So for a template pattern to not match, you want it to fail?  Should that be a switch?  Should the templater just fill in with a value of null?

**Comment by WesTyler** [comment]
@lamchakchan I'm not following :P
The underlying templating engine (doT in this case, but it looks like nunjucks may do the same thing) is improperly stringifying nested objects as `[object Object]`. That's the problem here. :)

**Comment by lamchakchan** [comment]
When you are templating for `{{=it.foo}}` and the data object you are feeding it is `{ foo: { bob: 'bar' } }` and does not contain the hierarchy of `it.foo`.  Should the templater not just respond with `null`?

**Comment by WesTyler** [comment]
Oh, ok, I'm with you now. the `=it.` is the doT syntax. The `it` refers to the context you are passing in to hydrate with. So in the case of `{{=it.foo}}` , `it` is `{ foo: { bob: 'bar' } }`.

**Comment by WesTyler** [comment]
(We are looking at options other than doT now though. I don't love the `it.` abstraction for this reason exactly, and handling the nested objects is getting messier and messier)

**Comment by WesTyler** [comment]
Closed via #2
